bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitHeaterCoolerImpl.java

```diff
@@ -0,0 +1,193 @@
+        service.addOptionalCharacteristic(new TemperatureDisplayUnitCharacteristic(() -> getTemperatureDisplayUnit(),
+                (value) -> setTemperatureDisplayUnit(value), (callback) -> subscribeTemperatureDisplayUnit(callback),
+                () -> unsubscribeTemperatureDisplayUnit()));
+        getServices().add(service);
+    }
+
+    @Override
+    public CompletableFuture<Double> getCurrentTemperature() {
+        @Nullable
+        final DecimalType state = getStateAs(HomekitCharacteristicType.CURRENT_TEMPERATURE, DecimalType.class);
+        return CompletableFuture.completedFuture(state != null ? convertToCelsius(state.doubleValue()) : 0.0);
+    }
+
+    @Override
+    public CompletableFuture<Boolean> isActive() {
+        final @Nullable State state = getStateAs(HomekitCharacteristicType.ACTIVE_STATUS, OnOffType.class);
+        return CompletableFuture.completedFuture(state == OnOffType.ON);
+    }
+
+    @Override
+    public CompletableFuture<Void> setActive(final boolean state) {
+        final @Nullable SwitchItem item = getItem(HomekitCharacteristicType.ACTIVE_STATUS, SwitchItem.class);
+        if (item != null) {
+            item.send(OnOffType.from(state));
+        }
+        return CompletableFuture.completedFuture(null);
+    }
+
+    @Override
+    public CompletableFuture<CurrentHeaterCoolerStateEnum> getCurrentHeaterCoolerState() {
+        return CompletableFuture.completedFuture(getKeyFromMapping(CURRENT_HEATER_COOLER_STATE, currentStateMapping,
+                CurrentHeaterCoolerStateEnum.INACTIVE));
+    }
+
+    @Override
+    public CompletableFuture<TargetHeaterCoolerStateEnum> getTargetHeaterCoolerState() {
+        return CompletableFuture.completedFuture(
+                getKeyFromMapping(TARGET_HEATER_COOLER_STATE, targetStateMapping, TargetHeaterCoolerStateEnum.AUTO));
+    }
+
+    @Override
+    public CompletableFuture<Void> setTargetHeaterCoolerState(final TargetHeaterCoolerStateEnum state) {
+        final Optional<HomekitTaggedItem> characteristic = getCharacteristic(
+                HomekitCharacteristicType.TARGET_HEATER_COOLER_STATE);
+        if (characteristic.isPresent()) {
+            ((StringItem) characteristic.get().getItem()).send(new StringType(targetStateMapping.get(state)));
+        } else {
+            logger.warn("Missing mandatory characteristic {}",
+                    HomekitCharacteristicType.TARGET_HEATING_COOLING_STATE.getTag());
+        }
+        return CompletableFuture.completedFuture(null);
+    }
+
+    public CompletableFuture<TemperatureDisplayUnitEnum> getTemperatureDisplayUnit() {
+        return CompletableFuture
+                .completedFuture(getSettings().useFahrenheitTemperature ? TemperatureDisplayUnitEnum.FAHRENHEIT
+                        : TemperatureDisplayUnitEnum.CELSIUS);
+    }
+
+    public void setTemperatureDisplayUnit(final TemperatureDisplayUnitEnum value) throws Exception {
+        // TODO: add support for display unit change
```

Comment:
When do you plan to implement this?

Reply:
Im not sure what is the best approach here from the user perspective. 
currently the temperature units (C/F) can be set at binding settings level. one can change the temperature units in the binding configuration and this will apply to all thermostats and coolers/heaters.

however, in home app one can change the temperature units for each accessory independently.  
to implement this i see 2 options
1) a new openHAB item to store temperature units (as part of accessory group)
2) handle units internally in accessory class. i.e. one can change it from home app but not from openHAB

1) sounds too much to me. we have already a lot of items for different characteristics. and that one is probably one nobody needs. 
2) does not sounds like a proper solution as one cannot change units from openHAB

the current solution - global setting is imho the best one. 
how you see it? what is the approach in other bindings?

Reply:
I would vote for global setting.

Reply:
ok. global settings are already working. i removed TODO here.